mitm.go

```diff
@@ -115,7 +116,27 @@ func (p *Proxy) serveConnect(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Proxy) cert(names ...string) (*tls.Certificate, error) {
-	return genCert(p.CA, names)
+	var (
+		cache = map[string]tls.Certificate{} // names (joined with ,) -> cert
```

Comment:
Ok; weird as it feels quicker (placebo?), also I borrowed the concept from some code of @bmizerany's. 

I'll read up on how to do this properly, if you're down for using it? :+1: